### ISSUE TITLE
OpenAPI JAX-RS Incomplete subresource support

### ISSUE THREAD

**Issue by vassterg** [payara/Payara#3042]
# Description #
----------
Incomplete support of openapi when subresource is present. 
I have two classes Foo and Bar such that Foo is a jaxrs resource and Bar is a subresource.
When deployed, the openapi link is incomplete.

## Expected Outcome

I would expect in the openapi link, the Bar sub resource GET endpoint to show up in the openapi and the Bar sub resource to be included as a schema.

## Current Outcome
```
openapi: 3.0.0
info:
  title: Deployed Resources
  version: 1.0.0
servers:
- url: http://localhost:8080/openapi-test
  description: Default Server.
paths:
  /resources/foo:
    get:
      operationId: getMethod
      responses:
        default:
          description: Default Response.
          content:
            '*/*':
              schema:
                type: string
components: {}
```
No endpoint and schema for subresource bar.

## Steps to reproduce (Only for bug reports) 
Create a simple javaee 8 maven JAX-RS project. 
Example pom.xml
```xml
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.insurit</groupId>
    <artifactId>openapi-test</artifactId>
    <version>1.0-SNAPSHOT</version>
    <packaging>war</packaging>
    <dependencies>
        <dependency>
            <groupId>javax</groupId>
            <artifactId>javaee-api</artifactId>
            <version>8.0</version>
            <scope>provided</scope>
        </dependency>
    </dependencies>
    <build>
        <finalName>openapi-test</finalName>
    </build>
    <properties>
        <maven.compiler.source>1.8</maven.compiler.source>
        <maven.compiler.target>1.8</maven.compiler.target>
        <failOnMissingWebXml>false</failOnMissingWebXml>
    </properties>
    <name>openapi-test</name>
</project>
```
Add the following three classes:
```java
@ApplicationPath("resources")
public class JAXRSConfiguration extends Application {

}
```
```java
@Path("foo")
public class Foo {

    @Context
    private ResourceContext rc;

    @GET
    public String getMethod() {
        return "get foo";
    }
    
    @Path("bar")
    public Bar sub() {
        return rc.getResource(Bar.class);
    }
}
```
```java
public class Bar {

    @GET
    public Response barMethod() {
        return Response.ok().entity("bar").build();
    }
}
```
Build and deploy project. Then access http://localhost:8080/openapi

## Environment ##

- **Payara Version**: 5.182
- **Edition**: Full
- **JDK Version**: 1.8.0_181; Java HotSpot(TM) 64-Bit Server VM 25.181-b13
- **Operating System**: Linux / Ubuntu 18.04
- **Database**:

**Comment by vassterg** [comment]
Attached is a sample pmaven project and the final artifact for deployment

[openapi-jaxrs-subresource-incomplete.zip](https://github.com/payara/Payara/files/2322703/openapi-jaxrs-subresource-incomplete.zip)
[openapi-incomplete.zip](https://github.com/payara/Payara/files/2322708/openapi-incomplete.zip)

**Comment by MattGill98** [comment]
Hi @vassterg,

The current OpenAPI implementation doesn't support subresources. I've made an enhancement request `PAYARA-3086` to add this functionality.

Kind regards,

Matt

**Comment by smillidge** [comment]
@jbee this is an old one do you think it will still be a problem?

**Comment by jbee** [comment]
@smillidge I believe we fixed this but I have to check.

**Comment by jbee** [comment]
@smillidge , @vassterg I was confusing this with another issue - @MattGill98  was and is correct in that we do not support subresources yet.

**Comment by rdebusscher** [comment]
Support for subresources is available in 5.194.
